crates/ruff/src/checkers/ast/analyze/scopes.rs

```diff
@@ -0,0 +1,299 @@
+                        if !matches!(
+                            shadowed.kind,
+                            BindingKind::Import(..)
+                                | BindingKind::FromImport(..)
+                                | BindingKind::SubmoduleImport(..)
+                                | BindingKind::FutureImport
+                        ) {
+                            continue;
+                        }
+                    }
+
+                    // If the bindings are in different forks, abort.
+                    if shadowed.source.map_or(true, |left| {
+                        binding.source.map_or(true, |right| {
+                            branch_detection::different_forks(left, right, &checker.semantic.stmts)
+                        })
+                    }) {
+                        continue;
+                    }
+
+                    #[allow(deprecated)]
+                    let line = checker.locator.compute_line_index(shadowed.range.start());
+                    let mut diagnostic = Diagnostic::new(
+                        pyflakes::rules::RedefinedWhileUnused {
+                            name: (*name).to_string(),
+                            line,
+                        },
+                        binding.range,
+                    );
+                    if let Some(range) = binding.parent_range(&checker.semantic) {
+                        diagnostic.set_parent(range.start());
+                    }
+                    diagnostics.push(diagnostic);
+                }
+            }
+        }
+
+        if matches!(
+            scope.kind,
+            ScopeKind::Function(_) | ScopeKind::AsyncFunction(_) | ScopeKind::Lambda(_)
+        ) {
+            if checker.enabled(Rule::UnusedVariable) {
+                pyflakes::rules::unused_variable(checker, scope, &mut diagnostics);
+            }
+
+            if checker.enabled(Rule::UnusedAnnotation) {
+                pyflakes::rules::unused_annotation(checker, scope, &mut diagnostics);
+            }
+
+            if !checker.is_stub {
+                if checker.any_enabled(&[
+                    Rule::UnusedFunctionArgument,
+                    Rule::UnusedMethodArgument,
+                    Rule::UnusedClassMethodArgument,
+                    Rule::UnusedStaticMethodArgument,
+                    Rule::UnusedLambdaArgument,
+                ]) {
+                    flake8_unused_arguments::rules::unused_arguments(
+                        checker,
+                        scope,
+                        &mut diagnostics,
+                    );
+                }
+            }
+        }
+
+        if matches!(
+            scope.kind,
+            ScopeKind::Function(_) | ScopeKind::AsyncFunction(_) | ScopeKind::Module
+        ) {
+            if enforce_typing_imports {
+                let runtime_imports: Vec<&Binding> = if checker.settings.flake8_type_checking.strict
```

Comment:
Can this be moved inside the second check below ? Doesn't seem to be used in the other rule